ipld/merkledag/merkledag.go

```diff
@@ -19,16 +18,14 @@ import (
 	basicnode "github.com/ipld/go-ipld-prime/node/basic"
 )
 
-// TODO: We should move these registrations elsewhere. Really, most of the IPLD
-// functionality should go in a `go-ipld` repo but that will take a lot of work
-// and design.
-func init() {
-	format.Register(cid.DagProtobuf, DecodeProtobufBlock)
-	format.Register(cid.Raw, DecodeRawBlock)
-	format.Register(cid.DagCBOR, ipldcbor.DecodeBlock)
+var ipldLegacyDecoder *legacy.Decoder
 
-	legacy.RegisterCodec(cid.DagProtobuf, dagpb.Type.PBNode, ProtoNodeConverter)
-	legacy.RegisterCodec(cid.Raw, basicnode.Prototype.Bytes, RawNodeConverter)
+// TODO: Don't require global registries
+func init() {
+	d := legacy.NewDecoder()
```

Comment:
@willscott WDYT about explicitly passing the go-ipld-prime linksystem here instead of requiring the default? This might make it easier for people to get out of the "global defaults" game.

Reply:
Initially it was just a stylistic preference that I had since a) it's still a global in 104, it's just a little more subtle b) it's a bit of a DRY thing where I didn't want to accidentally change one instead of the other.

However, I'm now also thinking that given we still have this global we might want to make it settable until we can make relying on a merkledag global registry no longer a thing